caffe2/CMakeLists.txt

```diff
@@ -497,7 +501,9 @@ if (NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
     install(TARGETS caffe2_nvrtc DESTINATION "${TORCH_INSTALL_LIB_DIR}")
   endif()
 
-  if (NOT NO_API)
+  if (INTERN_BUILD_MOBILE) 
```

Comment:
do we need both INTERN_BUILD_MOBILE and NO_API here? can we use one and drop the other? e.g.:
option 1: keep NO_API and set it automatically in master CMakeLists.txt for mobile build; use if (NOT NO_API) ... else() ...;
option 2: drop NO_API and use if (INTERN_BUILD_MOBILE) ... else () ...;

"csrc/api/src/jit.cpp" is already listed on line 432, so we should be able to directly set NO_API in master CMakeLists for mobile build.

Reply:
And can we make NO_API change a separate PR our rename this PR to better reflect your change.

You can use ghstack to maintain a stack of local diffs / PRs.

Reply:
There is one file - `output-archive.cpp` that depends on `export.h` is included via this `NO_API` flag. 


> And can we make NO_API change a separate PR our rename this PR to better reflect your change.

Yea, let's open another PR for this `NO_API` change

> do we need both INTERN_BUILD_MOBILE and NO_API here? can we use one and drop the other? e.g.:
> option 1: keep NO_API and set it automatically in master CMakeLists.txt for mobile build; use if (NOT NO_API) ... else() ...;
> option 2: drop NO_API and use if (INTERN_BUILD_MOBILE) ... else () ...;
> 
> "csrc/api/src/jit.cpp" is already listed on line 432, so we should be able to directly set NO_API in master CMakeLists for mobile build.

Actually I'm thinking if we can get rid of this `jit.cpp`, cuz there is only one function in the file, which does the compilation for cpp code. I've searched this symbol in the codebase, looks like we can safely remove it.